tests/unit/plugins/inventory/test_flightctl.py

```
@@ -0,0 +1,1112 @@
+        )
+
+        # Verify switch group contains only the switch device
+        self.assertIn('switches', self.inventory.inventory_data['groups'])
+        self.assertIn(
+            'device-456',  # The switch device
+            self.inventory.inventory_data['groups']['switches']['hosts']
+        )
+
+    @patch('plugins.inventory.flightctl._get_devices_by_labels_and_fields')
+    @patch('plugins.inventory.flightctl._prepare_additional_groups_info')
+    def test_populate_inventory_additional_groups(self, mock_prepare_additional_groups_info,
```

Ruff (0.8.2)

Redefinition of unused `test_populate_inventory_additional_groups` from line 329

(F811)